crest/Assets/Crest/Crest/Scripts/Helpers/Editor/EmbeddedAssetHelpers.cs

```diff
@@ -200,6 +211,10 @@ public void UpdateEditor(SerializedProperty property)
             if (m_Editor == null && target != null)
             {
                 m_Editor = Editor.CreateEditor(target);
+
+                // Pass through argument for editors that receive it
+                (m_Editor as IEditorWithArg)?.SetArgument(argument);
```

Comment:
I think we could change this to accepting a type. And then getting the type like so: `property.serializedObject.targetObject.GetType()`. And maybe rename IEditorWithArg to IEmbeddable. And SetArgument to SetEmbedderType? That way it is more clear what the pattern is for. If this whole thing is temporary then we can leave as is but would be nice if it was more clear if it does stick around. LMKWYT.

Reply:
ah yes much better! one